api/services/S3SiteRemover.js

```diff
@@ -38,33 +38,22 @@ const deleteObjects = (objects) => {
       }
     });
   }).then(() =>
-    deleteObjects(objectsToDeleteLater)
+    deleteObjects(keysToDeleteLater)
   );
 };
 
-const getObjectsWithPrefix = prefix =>
-  new Promise((resolve, reject) => {
-    s3Client.listObjects({
-      Bucket: s3Config.bucket,
-      Prefix: prefix,
-    }, (err, data) => {
-      if (err) {
-        reject(err);
-      } else {
-        const keys = data.Contents.map(object => object.Key);
-        resolve(keys);
-      }
-    });
-  });
+const getKeys = prefix =>
+  S3Helper.listObjects(prefix)
```

Comment:
Using the new helper here should mitigate https://github.com/18F/federalist/issues/1416